Addition of HTPC Manager

sytone opened the pull request:
Addition of HTPC Manager to make management better of different HTPC tools. 

Reversed out the process for a new app, hopefully it is correct! :)

sytone committed:
Addition of HTPC Manager

Fourdee commented on dietpi/dietpi-software:
```diff
@@ -1076,6 +1076,16 @@ _EOF_
 			 aSOFTWARE_ONLINEDOC_URL[$index_current]='f=8&t=5&p=7575#p7575'
 
+		#------------------
+		index_current=155
```
This is the index assigned to the software title, it must be used for all entries of:
```INSTALLING_INDEX=```

sytone replied:
Should be 155 everywhere. It was 152 when I was testing. Fixing for next iteration.

Fourdee commented on dietpi/dietpi-software:
```diff
@@ -6649,6 +6659,34 @@ _EOF_
 		fi
 
+		#HTPC Manager
+		INSTALLING_INDEX=152
```
Needs to match original index:
```INSTALLING_INDEX=155```

sytone replied:
Fixed.

Fourdee commented on dietpi/dietpi-software:
```diff
@@ -6649,6 +6659,34 @@ _EOF_
+				#Install Python and PIP
+				AGI python python-pip python-imaging
+
+				cd "$HOME"
```
Not required as this is called before the start of all software installs, however, no harm.

sytone replied:
The review is not showing the line not required. is it banner?

Fourdee replied:
@sytone 

```
cd "$HOME"
```

Although, no harm.

Fourdee commented on dietpi/dietpi-software:
```diff
@@ -6704,6 +6742,36 @@ _EOF_
 
+
+		#HTPC Manager
+		INSTALLING_INDEX=200
```
Needs to match original index:
```INSTALLING_INDEX=155```

Fourdee commented on dietpi/dietpi-software:
```diff
@@ -11022,6 +11090,27 @@ _EOF_
 		fi
 
+		#HTPC Manager
+		INSTALLING_INDEX=152
```
Needs to match original index:
```INSTALLING_INDEX=155```

sytone replied:
Fixed.

Fourdee commented on dietpi/dietpi-software:
```diff
@@ -11061,6 +11150,27 @@ WantedBy=multi-user.target
 		fi
+
+		#HTPC Manager
+		INSTALLING_INDEX=200
```
Needs to match original index:
```INSTALLING_INDEX=155```

sytone replied:
removed.

Fourdee commented on dietpi/dietpi-software:
```diff
@@ -11484,6 +11594,14 @@ _EOF_
 			rm -R "$FP_DIETPI_USERDATA_DIRECTORY"/nzbget
 			rm /etc/systemd/system/nzbget.service
 
+		elif (( $1 == 152 )); then
```
Needs to match original index:
```INSTALLING_INDEX=155```

eg:
```
elif (( $1 == 155 )); then
```

Fourdee commented on dietpi/dietpi-software:
```diff
@@ -11022,6 +11090,27 @@ _EOF_
+[Install]
+WantedBy=multi-user.target
+_EOF_
+			systemctl daemon-reload
```
Not required as both these entries are called at the end of software installations, however, no harm.

sytone replied:
Removed.

Fourdee commented on dietpi/dietpi-software:
```diff
@@ -11061,6 +11150,27 @@ WantedBy=multi-user.target
 
 		fi
+
+		#HTPC Manager
```
Duplicate code at line 11093.

sytone replied:
Removed

Fourdee commented on dietpi/dietpi-software:
```diff
@@ -6704,6 +6742,36 @@ _EOF_
 		fi
 
+
+		#HTPC Manager
```
Duplicate code at line 6662

sytone replied:
Removed

Fourdee commented:
@sytone 

Good effort 👍 

There are a few issues with index number assigned to the installation, needs to match ```155``` throughout. Also, appears to be duplicate install and configuration sections (2 of each).

I've made comments at the sections that need changes:
https://github.com/Fourdee/DietPi/pull/1083/files

Fourdee commented:
@sytone 
Also, we need to increase the max array size by 1 for software titles to ```156```, this will ensure your entry ```155``` exists:
```TOTAL_SOFTWARE_INDEXS_HARDLIMIT=156``` inside ```dietpi-software```

Fourdee commented on dietpi/dietpi-software:
```diff
@@ -11484,6 +11594,14 @@ _EOF_
+
+			rm -R "$FP_DIETPI_USERDATA_DIRECTORY"/htpc-manager
+			
+			systemctl disable htpc-manager.service			
```
```systemctl disable htpc-manager.service	```
Not required, as ```dietpi-services``` will disable this service during install (to take control), however, no harm.

sytone replied:
Removed.

Fourdee commented on dietpi/dietpi-software:
```diff
@@ -11484,6 +11594,14 @@ _EOF_
+			
+			systemctl disable htpc-manager.service			
+			rm /etc/systemd/system/htpc-manager.service
+			systemctl daemon-reload
```
```systemctl daemon-reload```
Not required, as ```systemctl daemon-reload``` is called at the end of uninstall script. However, no harm.

sytone replied:
Removed.

sytone committed:
Review Update

Resolved issues highlighted in Pull request.

sytone committed:
Pull Request Update

Fixed TOTAL_SOFTWARE_INDEXS_HARDLIMIT to be 156 to allow addition of HTPC Manager in UI.

sytone commented:
@Fourdee fixed the issues you highlighted, let me know if there are any other problems. I made a issue with a suggestion to separate these out to separate files to make this process simpler given the amount in the file now.

Fourdee commented:
@sytone 

>  fixed the issues you highlighted

Great work, looks good 👍 

We'll get this merged, i'll run a test in the next few days.

Fourdee reviewed (commented):
Completed.

sytone commented:
Thanks, when I have my account approved on the board I will do a write up for it.

Fourdee commented:
@sytone 

> Thanks, when I have my account approved on the board I will do a write up for it.

Done
